pgdog/src/frontend/router/parser/query.rs

```diff
@@ -701,6 +726,30 @@ impl QueryParser {
     fn delete(_stmt: &DeleteStmt) -> Result<Command, Error> {
         Ok(Command::Query(Route::write(None)))
     }
+
+    /// Check if a SELECT statement contains any functions with side effects.
+    fn has_side_effect_function(stmt: &SelectStmt) -> bool {
+        for target_item_wrapper_node in &stmt.target_list {
+            if let Some(NodeEnum::ResTarget(res_target)) = target_item_wrapper_node.node.as_ref() {
```

Comment:
`ResTarget` is only one possible token here, you can also have `TypeCast` and I think some others, e.g.:

```
SELECT pg_advisory_lock('test')::bool
```

Reply:
Fair point. Let me handle the other cases as well